exchanges/alert/alert.go

```diff
@@ -1,23 +1,50 @@
 package alert
 
 import (
+	"errors"
 	"sync"
 	"sync/atomic"
 )
 
-// pool is a silent shared pool between all notice instances for alerting
-// external routines waiting on a state change.
-var pool = sync.Pool{New: func() interface{} { return make(chan bool) }}
-
 const (
 	inactive = uint32(iota)
 	active
 	alerting
 
 	dataToActuatorDefaultBuffer = 1
-	preAllocCommsDefaultBuffer  = 5
+	PreAllocCommsDefaultBuffer  = 5
```

Comment:
Internalise. I know you're trying to set this via startup, but you don't want to introduce races here by having it exported that anything can change it at anytime. You have the `SetPreAllocationCommsBuffer` function to cover this

Reply:
This is a const and I haven't exported the `var preAllocBufferSize` either.